content/stellar-contracts/tokens/rwa/rwa.mdx

```diff
@@ -0,0 +1,512 @@
+
+Validates cryptographic claims and provides comprehensive claim lifecycle management. The module includes:
+
+**Signature Verification:**
+- Ed25519 (Stellar native)
+- Secp256k1 (Ethereum compatible)
+- Secp256r1 (Enterprise PKI compatible)
+
+**Key Management:**
+- Topic-specific key authorization with registry tracking
+- Each public key is tied to a signature scheme
+- A signing key (public key + scheme) can be authorized to sign claims for specific topic and registry combinations
+- The same signing key can be authorized across multiple topics and registries independently
+
+**Claim Invalidation Mechanisms:**
+- **Passive Expiration**: Helper functions to encode/decode expiration metadata (`created_at` and `valid_until` timestamps) within claim data, allowing claims to automatically expire without on-chain action
+- **Per-claim Revocation**: Fine-grained revocation of individual claims for precise control
+- **Signature Invalidation**: Efficient bulk invalidation via nonce increment for revoking all claims signed by a specific key
+
+## Extensions
+
+The following optional extensions are provided:
+
+### - Document Manager
+[Source Code](https://github.com/OpenZeppelin/stellar-contracts/tree/main/packages/tokens/src/rwa/extensions/doc_manager)
+
+This module is not mandatory.
+
+The `DocumentManager` trait extends the `RWAToken` trait to provide document management capabilities following the
+ERC-1643 standard. This extension allows contracts to:
+
+- Attach documents with URI, hash, and timestamp
+- Update existing document metadata
+- Remove documents from the contract
+- Retrieve individual or all documents
+
+This is useful for attaching legal documents, prospectuses, or other regulatory disclosures to the token contract.
+
+## Utility Modules
+
+### - Token Binder
+[Source Code](https://github.com/OpenZeppelin/stellar-contracts/tree/main/packages/tokens/src/rwa/utils/token_binder)
+
+The `TokenBinder` trait provides a standardized interface for linking tokens to periphery contracts such as identity
+registries and compliance contracts. This allows a single periphery contract to serve multiple RWA tokens.
+
+Features:
+- Bind/unbind tokens to periphery services
+- Query all linked tokens
+- Efficient storage with bucket-based architecture (supports up to 10,000 tokens)
+- Swap-remove pattern for compact storage
+
+## Security Considerations
+
+When implementing RWA tokens, consider the following security aspects:
+
+1. **Authorization**: All administrative functions expect RBAC checks to be enforced on the `operator` parameter
+2. **Identity Verification**: Always verify identities before allowing token operations
+3. **Compliance Validation**: Ensure compliance rules are properly configured before enabling transfers
+4. **Freezing**: Use freezing mechanisms carefully as they can lock user funds
+5. **Recovery**: Recovery operations should have strict authorization and verification
```

Comment:
maybe it's worth mentioning here that there's a 2-step process that must be handled with extra care instead of this generic phrase